pkg/serverless/trace/span_modifier.go

```diff
@@ -19,4 +21,14 @@ func (s *spanModifier) ModifySpan(span *pb.Span) {
 		// service name could be incorrectly set to 'aws.lambda' in datadog lambda libraries
 		span.Service = s.tags["service"]
 	}
+	if inferredspan.Check(span) {
+		log.Debug("Detected a managed service span, filtering out function tags")
+
+		// filter out existing function tags inside span metadata
+		spanMetadataTags := span.Meta
+		if spanMetadataTags != nil {
+			inferredspan.FilterFunctionTags(spanMetadataTags)
+			span.Meta = spanMetadataTags
```

Comment:
Per my other comment, I think this isn't quite right:
- Get the metadata tags `span.Meta`
- Call `FilterFunctionTags` on `spanMetadataTags`. This returns something (which is ignored here) and *also* modifies the original `spanMetadataTags`
- Set span.Meta to `spanMetadatatags` -- but I don't think this actually doing anything, it was already set to that. The map has already been updated in place.

Reply:
Yeah, this was working because of `output = input`, and originally the function being written in a way where `input` was directly modified.